webhub/tests.py

```diff
@@ -1,8 +1,341 @@
-#Version : Phython/Django 2.7.6, PostgreSQL 9.3.4
-#Author : Vaibhavi Desai
-#Github username : desaivaibhavi
-#email : ranihaileydesai@gmail.com
+from datetime import datetime
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.utils.six import BytesIO
+from rest_framework import status
+from rest_framework.parsers import JSONParser
+from rest_framework.renderers import JSONRenderer
+from rest_framework.test import APITestCase
+from webhub.models import Pcuser, Post
+from webhub.serializers import PostSerializer
 
-from django.test import TestCase
 
-# Create your tests here.
+class PostAPITestCase(APITestCase):
+
+    def setUp(self):
+        """Setup the test database, test data and authenticate"""
+
+        u1 = User.objects.create_superuser(username='admin',
+                                           password='password', email='')
+        u1.save()
+
+        o1 = Pcuser(user=u1)
+        o1.save()
+
+        p1 = Post(owner=o1,
+                  title_post="Title 1",
+                  description_post="Description 1")
+
+        p2 = Post(owner=o1,
+                  title_post="Title 2",
+                  description_post="Description 2")
+
+        p3 = Post(owner=o1,
+                  title_post="Title 3",
+                  description_post="Description 3")
+
+        p1.save()
+        p2.save()
+        p3.save()
+
+        self.data_1 = {'owner': 1,
+                       'title_post': 'Test 1',
+                       'description_post': 'Test 1',
+                       'created': datetime.now(),
+                       'updated': datetime.now(),
+                       'id': '1'}
+
+        self.data_2 = {'owner': 1,
+                       'title_post': 'Test 2',
+                       'description_post': 'Test 2',
+                       'created': datetime.now(),
+                       'updated': datetime.now(),
+                       'id': '2'}
+
+        self.data_3 = {'owner': 1,
+                       'title_post': 'Test 3',
+                       'description_post': 'Test 3',
+                       'created': datetime.now(),
+                       'updated': datetime.now(),
+                       'id': '3'}
+
+        self.authenticate()
+
+    def authenticate(self):
```

Comment:
can we add some failure test cases .. like incorrect pwd/username and asserting we get the response we meant to show?